workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/ManifestContextBuilder.java

```diff
@@ -54,14 +54,14 @@ public ManifestContext build(WorkflowContext workflowContext) throws HyscaleExce
         manifestContext.setImageRegistry(registryManager.getImageRegistry(registryUrl));
         manifestContext.addGenerationAttribute(ManifestGenConstants.IMAGE_SHA_SUM,
                 workflowContext.getAttribute(WorkflowConstants.IMAGE_SHA_SUM));
-        Map<String, String> label = getLabel(workflowContext);
-        if (label != null && !label.isEmpty()) {
-            manifestContext.putCustomLabel(label);
+        Map<String, String> customLabels = getCustomLabels(workflowContext);
+        if (customLabels != null && !customLabels.isEmpty()) {
```

Comment:
In the method below, we can return an empty map, that will take care of major bug of the condition always evaluating as false